INTEGRATIONS.md

```diff
@@ -0,0 +1,92 @@
+# Gas Price Queries for Integrations
+
+Because `x/feemarket` uses a dynamic fee, end-users will need to query the module for the current `gasPrice` to use when building a transaction.
+
+A summary for the flow to query this information is as follows:
+
+* Create an RPC connection with the chain
+* Create a `feemarket` client
+* Submit the `GasPrice` query 
+* Use the `gasPrice` to populate the Tx fee field.
+
+Extensive querying information can be seen in the module [spec](./README.md#query).
+
+The specific query for `GasPrices` can be found [here](./README.md#gas-prices).
+
+## Code Snippet
+
+Wallet, relayers, and other users will want to add programmatic ways to query this before building their transactions.  Below is an example of how a user could implement this lightweight query in Go:
+
+### Create A gRPC Connection
+
+First, a base connection to the chain you are querying must be created.
+
+A chain gRPC (below) or CometBFT ABCI RPC connection can be created:
+
+```go
+	// Set up gRPC connection to chain
+   cc, err := grpc.NewClient(endpoint, insecure.NewCredentials())
+   if err != nil {
+	   panic(err)
+   }
+   defer cc.Close()
+```
+
+### Create a FeeMarket Query Client
+
+An `x/feemarket` query client can then be created using the created gRPC connection.
+
+This client exposes all [queries](./README.md#query) that the `x/feemarket` module exposes.
+
+```go
+   // Create FeeMarketClient with underlying gRPC connection
+   feeMarketClient := feemarkettypes.NewQueryClient(cc)
+```
+
+### Query Gas Prices 
+
+The `gas price` (as an `sdk.DecCoin`) can be queried using the `GasPrice` query.  This query requires the desired coin denomination for the fee to be paid with.
+
+The query will return an error if the given denomination is not supported.
+
+```go
+   gasPrice, err := feeMarketClient.GasPrice(ctx, &feemarkettypes.GasPriceRequest{
+	   Denom: denom,
+   })
+   if err != nil {
+	   panic(err)
+   }
+```
+
+### Using Gas Price
+
+The calculation for a transaction's fee is: `gasLimit` * `gasPrice`.  End users can simulate a Tx or use a fixed gas limit as they would have done before when setting the `fee` on their transactions.
+
+The actual amount that will be deducted from a user's account is:
+
+```
+  deducted = gasConsumed * gasPrice
+```
+
+Users can also optionally add a `tip` to their transaction which will be paid to the block proposer as well as increase their transactions' priority.  In Cosmos, the priority of your Tx is the entire `fee` provided.  Transactions with higher fees are always ordered higher than others with lower fees in blocks.
+
+A fee with a tip is as follows:
+
+```
+  feeWithTip = tip + (gasLimit * gasPrice)
+```
+
+The deducted amount after a transaction will be:
+
+```
+  deducted = tip + (gasConsumed * gasPrice)
```

Comment:
### Using `gasPrice` to construct a transaction
There are two ways to construct a transaction with `gasPrice`: 
1.  Provide the minimum fee: `feeAmount = gasPrice * gasLimit` (`gasLimit` gives the maximum amount of gas a transaction can consume. You can obtain appropriate `gasLimit` by simulating a transaction to see how much gas it consumes under normal conditions).
2. Provide a "tip" in addition to the minimum fee: `feeAmount=gasPrice * gasLimit + tip` This will be paid to the block proposer and result in your transaction being placed ahead of others with lower tips (or being included in the block instead of others when the block is full) 

### Understanding Fee Deducted
The actual amount of fee deducted from the fee payer is based on gas consumed, not `gasLimit`. 

The amount consumed is equal to the `inferredTip + gasPrice * gasConsumed`, where `inferredTip = feeAmount - gasLimit * gasPrice`  (This may be different than the tip you specified when building the transaction because the `gasPrice` on chain may have changed since when you queried it.)